src/org/thoughtcrime/securesms/LocalHelpActivity.java

```diff
@@ -1,269 +1,80 @@
-                    updateResultCounter(-1, 0); // hide
-                } else {
-                    String msg = getString(R.string.search_no_result_for_x, lastQuery);
-                    if (lastToast != null) {
-                        lastToast.cancel();
-                    }
-                    lastToast = Toast.makeText(this, msg, Toast.LENGTH_SHORT);
-                    lastToast.show();
-                    updateResultCounter(0, 0); // show as "0/0"
-                }
-            }
-        }
-    }
-
-
-    // other actions
-
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        super.onOptionsItemSelected(item);
-        switch (item.getItemId()) {
-            case android.R.id.home:
-                finish();
-                return true;
-            case R.id.menu_search_up:
-                if (lastQuery.isEmpty()) {
-                    webView.scrollTo(0, 0);
-                } else {
-                    webView.findNext(false);
-                }
-                return true;
-            case R.id.menu_search_down:
-                if (lastQuery.isEmpty()) {
-                    webView.scrollTo(0, 1000000000);
-                } else {
-                    webView.findNext(true);
-                }
-                return true;
-            case R.id.log_scroll_up:
-                webView.scrollTo(0, 0);
-                return true;
-            case R.id.learn_more:
-                openOnlineUrl("https://delta.chat");
-                return true;
-            case R.id.contribute:
-                openOnlineUrl("https://github.com/deltachat/deltachat-android");
-                return true;
-            case R.id.report_issue:
-                openOnlineUrl("https://github.com/deltachat/deltachat-android/issues");
-                return true;
-        }
-        return false;
-    }
-
-    @Override
-    public void onBackPressed() {
-        if (webView.canGoBack()) {
-            webView.goBack();
-        } else {
-            super.onBackPressed();
-        }
-    }
-
-    private void openOnlineUrl(String url) {
-        try {
-            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(url)));
-        } catch (ActivityNotFoundException e) {
-            Toast.makeText(this, R.string.no_browser_installed, Toast.LENGTH_LONG).show();
-        }
-    }
-
-    @Override
-    public void onActivityResult(int reqCode, int resultCode, final Intent data) {
-        super.onActivityResult(reqCode, resultCode, data);
+    return false;
+  }
+
+  private boolean assetExists(String fileName) {
+    // test using AssetManager.open();
+    // AssetManager.list() is unreliable eg. on my Android 7 Moto G
+    // and also reported to be pretty slow.
+    boolean exists = false;
+    try {
+      AssetManager assetManager = getResources().getAssets();
+      InputStream is = assetManager.open(fileName);
+      exists = true;
+      is.close();
+    } catch(Exception e) {
+      ;
```

Comment:
Maybe forgot to print the exception? (if it happens from time to time and you don't want to spam the log, maybe also use `Log.w(TAG, e);`, this will only print just a single line)

Reply:
errors thrown in this function are normal an expected, we are using it to check for existance, as the "normal" ways to check for assets are erroneous. logging does not make much sense therefore, you would get warnings/info everytime you open the help. this code is also not changed by this pr, it was just copied here (sorry, i should have mention that: LocalHelpActivity was included in WebViewActivity before.

Reply:
but adding a comment makes sense, i do a commit.